apps/src/lib/levelbuilder/lesson-editor/AddLevelDialogTop.jsx

```diff
@@ -9,57 +9,38 @@ import queryString from 'query-string';
 import FontAwesome from '@cdo/apps/templates/FontAwesome';
 import {connect} from 'react-redux';
 
-class AddLevelDialogTop extends Component {
-  static propTypes = {
-    addLevel: PropTypes.func.isRequired,
-
-    // from redux
-    searchOptions: PropTypes.object.isRequired
-  };
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      methodOfAddingLevel: 'Find Level',
-      levels: [],
-      loadingLevels: true,
-      currentPage: 1,
-      levelName: '',
-      levelType: props.searchOptions.levelOptions[0][1],
-      unitId: props.searchOptions.scriptOptions[0][1],
-      ownerId: props.searchOptions.ownerOptions[0][1],
-      numPages: 0
-    };
-  }
-
-  componentDidMount() {
-    $.ajax({
-      url: `/levels/get_filtered_levels?page=${this.state.currentPage}`,
-      method: 'GET',
-      contentType: 'application/json;charset=UTF-8'
-    }).done(data => {
-      this.setState({
-        loadingLevels: false,
-        levels: data.levels,
-        numPages: data.numPages
-      });
-    });
-  }
-
-  handleSearch = () => {
-    let queryParams = {page: this.state.currentPage};
-    if (this.state.levelName) {
-      queryParams.name = this.state.levelName;
+function AddLevelDialogTop(props) {
+  const [methodOfAddingLevel, setMethodOfAddingLevel] = useState('Find Level');
+  const [levels, setLevels] = useState([]);
+  const [loadingLevels, setLoadingLevels] = useState(true);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [levelName, setLevelName] = useState('');
+  const [levelType, setLevelType] = useState(
+    props.searchOptions.levelOptions[0][1]
+  );
+  const [unitId, setUnitId] = useState(props.searchOptions.scriptOptions[0][1]);
+  const [ownerId, setOwnerId] = useState(
+    props.searchOptions.ownerOptions[0][1]
+  );
+  const [numPages, setNumPages] = useState(0);
+
+  useEffect(() => {
+    handleSearch();
+  }, [currentPage]);
+
+  const handleSearch = () => {
```

Comment:
So, technically, I think if we were to enable the eslint rules for hooks, it would have a warning about adding `handleSearch` to the `useEffect` dependency array (because you're supposed to add anything that could change and it changes on every render), but then the search would get called every render, so to get around that you'd want to wrap `handleSearch` in a `useCallback`

I am not entirely sure about all the nuances around why/when there might be an issue caused by this. I think there isn't actually any problem with this logic as is, because of how we only want the search to happen when the page changes (and all the params i assume are bits of state that are dependent on the page).

Something related to note here is that since `handleSearch` is a new function on each render, `AddLevelFilters` will be rerendered every time this parent component rerenders because its props will have changed. We can avoid that with `useCallback` also.

Reply:
or maybe there is actually a bug that would have been caught by the eslint rules and we should be calling handleSearch any time any of the query params change also!

Reply:
We call handleSearch from another component when the search button is clicked so I was going off the trigger of clicking that button for the other fields. We also debounce the clicking of the search in that component. I did explore the idea of removing the search button and just updating the levels shown as the dropdowns/inputs were changed. That would be a different user experience so I don't know how we feel about that.

Reply:
yeah, i think this works fine as is! i need to do some more research on how exactly the `exhaustive-deps` (where everything in the effect is supposed to be in the dependency array) rule works.

Reply:
ok i did a bit more reading about this and i think it might make more sense to move the handleSearch call into the event handler instead of using a `useEffect`, because here we're using the page change as a trigger but really its the event that should trigger the search and a page change just happens to happen at the same time?

Reply:
Ok I updated it so when we set the page number we call handle search. We still need to the useEffect for the initial load. Does that look ok?

Reply:
yeah! I think that makes more sense